web-ncer/app/views/sampleRequest/submissionForm.scala.html

```diff
@@ -0,0 +1,114 @@
+                Request Samples
+            </button>
+        </div>
+        <div class="form-group">
+            @catalogueItemDropdown
+        </div>
+        <div id="application-link" class="form-group">
+            <label for="remsLink">Direct Link to Application:</label><br/>
+            <a id="remsLink" href="" target="_blank"></a>
+        </div>
+        <p id="request-error" style="display: none;"></p>
+    </div>
+}
+
+@extraBottomResources = {
+    <script type="text/javascript">
+            $(function () {
+                // get all the filters available
+                var filters = $("#filtersTr").find(".filter-div")
+                $("#widgetsTr").html("")
+                $.each(filters, function (i, filter) {
+                    var td = $("<td style='vertical-align:top'>")
+                    var row = $("<div class='row'>")
+                    td.append(row)
+                    $("#widgetsTr").append(td)
+                })
+                // generate (refresh) a view for each filter
+                $.each(filters, function (index, filter) {
+                    $(filter).multiFilter('submitFilter');
+                })
+            })
+            function submitForm() {
+                $('#request-error').hide()
+                const option = JSON.parse($('#remsCatalogueItemDropdown').val())
+                const action = '@Html(sampleRequestRoutes.submitRequest(dataSetSetting.dataSetId).url)'
+                const table = getSelectedRowIds($('#tablesTr .table-div'))
+                const selectedIds = []
+                $.each(table, function (index, id) {
+                    selectedIds.push(id)
+                })
+                const params = {
+                    'tableColumnNames[]': '@tableViewParts.head.tableFields.map(_.name).toVector.mkString("\t")'.split("\t"),
+                    catalogueItemId: option.id,
+                    catalogueFormId: option.formId,
+                    'selectedIds[]': selectedIds,
+                    'conditions[]': $('.filter-div').multiFilter('getModel').map(model => JSON.stringify(model))
+                }
+
+                const form = $('<form></form>');
+                form.attr("method", "POST");
+                form.attr("action", action);
+                addParams(form, params)
+                $.ajax({
+                    type: form.attr('method'),
+                    url: form.attr('action'),
+                    data: form.serialize(),
+                    success: function(url) {
+                        const a = $('#remsLink')
+                        a.text(url)
+                        a.attr('href', url)
+                    },
+                    error: function(ctx) {
+                        const p = $('#request-error')
+                        p.text(ctx.responseText)
+                        p.show()
+                    }
+                });
```

Comment:
this entire thing can be replaced by a Play JsRouter... see e.g. `category/ist.scala.html`:

```
            categoryJsRoutes.org.ada.web.controllers.dataset.CategoryDispatcher.saveForName(name).ajax( {
                success: function(id) {
                    $('#treeDiv').tree('addToRoot', id, name);
                    showMessage("The category '" + name + "' successfully created.");
                },
                error: function(data){
                    showError( data.responseText );
                }
            });
```